File: ch11/yojkim/ch11/money_test.go
package ch11

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestCurrency(t *testing.T) {
	assert.Equal(t, "USD", NewMoney(5, "USD").Currency())
	assert.NotEqual(t, "CHF", NewMoney(5, "USD").Currency())
}

func TestMultiplication(t *testing.T) {
	dollar := NewMoney(5, "USD")
	assert.Equal(t, NewMoney(10, "USD"), dollar.Times(2))
	assert.Equal(t, NewMoney(15, "USD"), dollar.Times(3))
}

func TestEquality(t *testing.T) {
	assert.True(t, NewMoney(5, "USD").Equal(NewMoney(5, "USD")))
	assert.False(t, NewMoney(5, "USD").Equal(NewMoney(6, "USD")))
	assert.False(t, NewMoney(5, "USD").Equal(NewMoney(5, "CHF")))
}

File: ch11/gunddol/src/ch11/Money_Test.java
package ch11;

import org.junit.Test;

import static org.junit.Assert.*;

public class Money_Test {
    @Test
    public void testEquality(){
        assertTrue(Money.dollar(5).equals(Money.dollar(5)));
        assertFalse(Money.dollar(5).equals(Money.dollar(6)));
        assertFalse(Money.franc(5).equals(Money.dollar(5))); //금액과 클래스가 모두 같아야 참
    }

    @Test
    public void testCurrency(){
        assertEquals("USD",Money.dollar(1).currency());
        assertEquals("CHF",Money.franc(1).currency());
    }
}